### DISCUSSION TITLE
False Liter in Homeassistant

### DISCUSSION THREAD

**Discussion by 1990marco1990** [jomjol/AI-on-the-edge-device#1528] [Q&A]
`Hello
I have installed the watermeter and it looks good but the value of L ist very small because the comma ist not on the right point. Can anyone help me?

![Screenshot_2022-12-09-06-40-59-050_com sec android app sbrowser](https://user-images.githubusercontent.com/91014007/206636152-560da892-5d3e-4f94-8120-758b8966201c.jpg)
![Screenshot_2022-12-09-07-02-58-640_io homeassistant companion android](https://user-images.githubusercontent.com/91014007/206636160-08c6eb9a-eb50-4105-96bb-c86a939079cf.jpg)
`

**Comment by 1990marco1990** [comment]
Nobody a solution? If i Change the decimal it did only with the digital cointer not with the analog counter. Its a Bug?

**Comment by jomjol** [comment]
@caco3 : is this a topic of the home assist implementation?

**Comment by caco3** [comment]
It is a bit complicated:
The device itself does not know what kind of unit you use. Therefore the `Value` and `rate` are unit-less.
But since Homeassistant requires the value to have a unit (so the Energy Dashboard can use it), we have to add a unit.

Most watermeter provide `m3`, so you should set the meter type to use `m3`.
If you want to have the values in `l` instead, you have to:
 - set the meter type to use `l` **and**
 - set the `DecimalShift` to `-3` (or maybe  `3`, I did not test it)
